documentation/jetty/modules/programming-guide/pages/troubleshooting/component-dump.adoc

```diff
@@ -11,16 +11,106 @@
 // ========================================================================
 //
 
-= Component Tree Dump
+[[dump]]
+= Jetty Component Dump
 
-Jetty components are organized in a xref:arch/bean.adoc[component tree].
+Jetty is based on components organized as a tree of Objects (aka beans), with the `Server` or `Client` instance typically at the root of the tree.
 
-At the root of the component tree there is typically a `ContainerLifeCycle` instance -- typically a `Server` instance on the server and an `HttpClient` instance on the client.
+As explained in the xref:operations-guide:jmx/index.adoc[JMX section], these components can be exported as JMX MBeans and therefore be accessible from JMX Consoles such as Java Missions Control (JMC).
 
-`ContainerLifeCycle` has built-in _dump_ APIs that can be invoked either directly on the `Server` instance, or xref:arch/jmx.adoc[via JMX].
+Being able to take a snapshot of a Jetty component tree while it is running is the most useful information that can be attached when reporting an issue as it can contain both configuration and current state of Jetty.
+A dump of the component tree for the Server includes:
 
-You can invoke `Server.dump()` via JMX using a JMX console such as https://adoptium.net/jmc.html[Java Mission Control (JMC)]:
+* The thread pool configuration and its current state, including how many threads are in use, and their stack trace.
+* The Connector configuration (including TLS configuration), state and statistics.
+* The I/O configuration and its current state, including the ports Jetty listens to, how many connections are currently open, and he state of each connection, and the state of the request/response handling for each connection.
+* The `Handler` structure and its configuration.
+* The web applications deployed and their configurations, including the class loader information.
+
+[[dump-start-stop]]
+== Dump at Server Start/Stop
+
+The `Server.dump()` operation may also be invoked just after the `Server` starts (to log the state of the freshly started server), and just before the `Server` stops (which may be useful to log the state of server that is not working properly).
+
+You can temporarily enable the Jetty Server Dump at start time by overriding the `jetty.server.dumpAfterStart` property on the command line:
+
+For embedded usage this can be used by calling the setters directly.
+[,java,indent=0]
+----
+include::code:example$src/main/java/org/eclipse/jetty/docs/programming/ComponentDocs.java[tags=dumpExample]
+----
+
+Standalone Jetty uses two properties to control this behaviour which are referenced in `jetty.xml` to call these setters.
+These properties are `jetty.server.dumpAfterStart` and `jetty.server.dumpBeforeStop`.
```

Comment:
No, this is why previously the documentation was split.
This is the programming guide and referring to properties that are used only in the operations guide is not the correct place.